### PR TITLE
entrezdirect/xtract

### LINKED ISSUE #1832: new module: entrezdirect/xtract
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by alxndrdiaz**
## PR checklist

Closes #1832 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by muffato** [comment]
@alxndrdiaz : can you remove the _esummary_ files ?

**Comment by alxndrdiaz** [comment]
> @alxndrdiaz : can you remove the _esummary_ files ?

Yes, I included them because they are required to generate the text files for the tests, but another solution will be to uplodad the two text files to nf-core test datasests and fetch them from there.

**Review by jfy133 — COMMENTED** [review]
I'll wait until the main ENTREZDIRECT_ESUMMARY is merged in and updated here before reviewing :+1:

**Review by muffato — CHANGES_REQUESTED** [review]
As in #1833

**Review by muffato — CHANGES_REQUESTED** [review]
Last request: this pull-request shouldn't be touching `esummary`. Can you revert the three esummary files ?

File: modules/entrezdirect/esummary/main.nf
Comment (muffato) [inline modules/entrezdirect/esummary/main.nf:24]:
```suggestion
```

File: tests/config/pytest_modules.yml
Comment (muffato) [inline tests/config/pytest_modules.yml:692]:
```suggestion
entrezdirect/esummary:
  - modules/entrezdirect/esummary/**
  - tests/modules/entrezdirect/esummary/**
  
  entrezdirect/xtract:
  - modules/entrezdirect/xtract/**
  - tests/modules/entrezdirect/xtract/**
```

File: tests/modules/entrezdirect/esummary/main.nf
Comment (muffato) [inline tests/modules/entrezdirect/esummary/main.nf:12]:
```suggestion
```

File: tests/modules/entrezdirect/esummary/test.yml
Comment (muffato) [inline tests/modules/entrezdirect/esummary/test.yml:26]: Hopefully this will restore the final line return
```suggestion
      md5sum: 01c962bc82d3b3bcb9d4da5b0b6b408a
```

**Review by muffato — COMMENTED** [review]
This is what I meant: I'd prefer the changelog not to show any changes (even blank lines) in the esummary files / related to esummary.

File: modules/entrezdirect/xtract/main.nf
Comment (jfy133) [inline modules/entrezdirect/xtract/main.nf:3]: If I understand it this is just text manuulation, low would be sufficient I imaigne

File: modules/entrezdirect/xtract/main.nf
Comment (jfy133) [inline modules/entrezdirect/xtract/main.nf:31]:
```suggestion
        > ${prefix}.csv
```

File: modules/entrezdirect/xtract/main.nf
Comment (jfy133) [inline modules/entrezdirect/xtract/main.nf:16]:
```suggestion
    tuple val(meta), path("*.csv"), emit: xtract_table
```

File: modules/entrezdirect/xtract/main.nf
Comment (jfy133) [inline modules/entrezdirect/xtract/main.nf:28]: You should not hardcode parameters here. As it seems there isn't an obvious output file type, I would just set it as `txt`. This will allow others to set the separator type as they wish.

File: modules/entrezdirect/xtract/meta.yml
Comment (jfy133) [inline modules/entrezdirect/xtract/meta.yml:49]: See comments on `main.nf`

File: tests/modules/entrezdirect/xtract/test.yml
Comment (jfy133) [inline tests/modules/entrezdirect/xtract/test.yml:9]: File name will need to be updated

**Review by jfy133 — CHANGES_REQUESTED** [review]
Given you can transmit various output types, I wouldn't force users to `csv`. I would suggest making it generic by having the output as simply `.txt`

File: modules/entrezdirect/xtract/main.nf
Comment (alxndrdiaz) [reply modules/entrezdirect/xtract/main.nf:3]: True, in fact I forgot to change it.

**Review by alxndrdiaz — COMMENTED** [review]
Removed files from wrong branch merge.

**Comment by muffato** [comment]
Hi @alxndrdiaz . A lot more files appeared in the diff 😓 ! Maybe it'd be easier to start off the freshest ncfore master branch, and just commit your 5 new files + updated `tests/config/pytest_modules.yml` ?

**Pull request closed without merging**
